src/Interpreters/HashJoin.cpp

```diff
@@ -194,6 +193,13 @@ HashJoin::HashJoin(std::shared_ptr<TableJoin> table_join_, const Block & right_s
 
     required_right_keys = table_join->getRequiredRightKeys(right_table_keys, required_right_keys_sources);
 
+    LOG_DEBUG(log, "Right keys: [{}] (required: [{}]), left keys: [{}]",
+              fmt::join(key_names_right, ", "),
+              fmt::join(required_right_keys.getNames(), ", "),
+              fmt::join(table_join->keyNamesLeft(), ", "));
+
+    LOG_DEBUG(log, "Columns to add: [{}]", sample_block_with_columns_to_add.dumpStructure());
+
```

Comment:
Did you forget to remove it?

Reply:
I suppose we can leave it, it already prints 'right_sample_block' and sometimes it's good to see on what keys join performed and what columns will be added to left table.

Reply:
Ok, let's leave it